http.go

```diff
@@ -96,56 +91,32 @@ func probeHTTP(target string, w http.ResponseWriter, module Module, registry *pr
 	registry.MustRegister(redirectsGauge)
 	registry.MustRegister(isSSLGauge)
 	registry.MustRegister(statusCodeGauge)
-	registry.MustRegister(probeIPProtocolGauge)
 
 	config := module.HTTP
 
-	if module.HTTP.Protocol == "" {
-		module.HTTP.Protocol = "tcp"
-	}
-
-	if module.HTTP.Protocol == "tcp" && module.HTTP.PreferredIPProtocol == "" {
-		module.HTTP.PreferredIPProtocol = "ip6"
-	}
-	if module.HTTP.PreferredIPProtocol == "ip6" {
-		fallbackProtocol = "ip4"
-	} else {
-		fallbackProtocol = "ip6"
-	}
 	if !strings.HasPrefix(target, "http://") && !strings.HasPrefix(target, "https://") {
 		target = "http://" + target
 	}
 
-	dialProtocol = module.HTTP.Protocol
-	if module.HTTP.Protocol == "tcp" {
-		targetURL, err := url.Parse(target)
-		if err != nil {
-			return false
-		}
-		targetHost, _, err := net.SplitHostPort(targetURL.Host)
-		// If split fails, assuming it's a hostname without port part
-		if err != nil {
-			targetHost = targetURL.Host
-		}
-		ip, err := net.ResolveIPAddr(module.HTTP.PreferredIPProtocol, targetHost)
-		if err != nil {
-			ip, err = net.ResolveIPAddr(fallbackProtocol, targetHost)
-			if err != nil {
-				return false
-			}
-		}
+	targetURL, err := url.Parse(target)
+	if err != nil {
+		return false
+	}
+	targetHost, _, err := net.SplitHostPort(targetURL.Host)
+	// If split fails, assuming it's a hostname without port part.
+	if err != nil {
+		targetHost = targetURL.Host
+	}
 
-		if ip.IP.To4() == nil {
-			dialProtocol = "tcp6"
-		} else {
-			dialProtocol = "tcp4"
-		}
+	ip, err := chooseProtocol(module.HTTP.PreferredIPProtocol, targetHost, registry)
+	if err != nil {
+		return false
 	}
 
-	if dialProtocol == "tcp6" {
-		probeIPProtocolGauge.Set(6)
+	if ip == nil {
```

Comment:
Shouldn't this be checking the type of the ip? This is also not needed, the IP alone implies the protocol.